commandapi-platforms/commandapi-bukkit/commandapi-bukkit-test/commandapi-bukkit-test-tests/src/test/java/dev/jorel/commandapi/test/CommandRegistrationTests.java

```diff
@@ -707,4 +707,33 @@ void testCommandTreeDuplicateNodeNameException() {
 			getDispatcherString()
 		);
 	}
-}
+
+	@Test
+	void testCommandTreeThenNested() {
+		// Make sure dispatcher is cleared from any previous tests
+		CommandAPIHandler.getInstance().writeDispatcherToFile();
+
+		// Register a command using the legacy `then` method
+		new CommandTree("test1").then(
+			new LiteralArgument("a").then(
+				new LiteralArgument("b").then(
+					new LiteralArgument("c")
+						.executesPlayer(P_EXEC)
+				)
+			)
+		);
+		String dispatcherStrLegacy = getDispatcherString();
+
+		// Register a command using the new `thenNested` method
+		new CommandTree("test2").thenNested(
+			new LiteralArgument("a"),
+			new LiteralArgument("b"),
+			new LiteralArgument("c")
+				.executesPlayer(P_EXEC)
+		);
+		String dispatcherStrNested = getDispatcherString();
+
+		// Both commands should have the same dispatcher string
+		assertEquals(dispatcherStrLegacy, dispatcherStrNested);
```

Comment:
This test doesn't do what you think it does. We have this which writes the command dispatcher to a file:
```java
CommandAPIHandler.getInstance().writeDispatcherToFile();
```
Then we call `getDispatcherString()` (which returns the file contents)
Then we call `getDispatcherString()` again (which returns the same file contents)

We never actually update the file because you never call `.register()`. A better test would be to do something like splitting this into two tests: One to test that this generates an expected result (as a control test):
```java
new CommandTree("test1").then(
	new LiteralArgument("a").then(
		new LiteralArgument("b").then(
			new LiteralArgument("c")
				.executesPlayer(P_EXEC)
		)
	)
);
```
And another to test that this generates an expected (and identical) result:
```java
new CommandTree("test2").thenNested(
	new LiteralArgument("a"),
	new LiteralArgument("b"),
	new LiteralArgument("c")
		.executesPlayer(P_EXEC)
);
```

Reply:
is that work if I add `.register()` to both CommandTree? I thought the test will show that we expected these two CommandTree will be the same

Reply:
Now that I understand what you mean, I'm going to revise my test, thanks very much!